### ISSUE TITLE
pwnlib.util.net.sockaddr incorrect for IPv6

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#541]
`getaddrinfo` does not strictly respect the address family provided.  We need to filter the list based on the correct address family, rather than just taking the first entry.

```
socket.getaddrinfo('google.com',socket.AF_INET6)
[(2, 1, 6, '', ('173.194.196.100', 10)),
 (2, 2, 17, '', ('173.194.196.100', 10)),
 (2, 3, 0, '', ('173.194.196.100', 10)),
...
 (2, 3, 0, '', ('173.194.196.138', 10)),
 (10, 1, 6, '', ('2607:f8b0:4001:c06::8b', 10, 0, 0)),
 (10, 2, 17, '', ('2607:f8b0:4001:c06::8b', 10, 0, 0)),
 (10, 3, 0, '', ('2607:f8b0:4001:c06::8b', 10, 0, 0))]
```

**Comment by RobertLarsen** [comment]
It does that already:

    for family, _, _, _, ip in socket.getaddrinfo(host, None, address_family):
        ip = ip[0]
        print family, ip, address_family
        if family == address_family: <--right here
            break
    else:
        log.error("Could not find %s address for %r" % (network, host))

However, if the host has no IPv6 addresses it may end up using the last address in the list

**Comment by zachriggle** [comment]
I believe that's the situation that I ran into.

**Comment by RobertLarsen** [comment]
That is a problem...should probably log an error and return None instead...or raise an exception. Not sure.

**Comment by RobertLarsen** [comment]
Hmm...my domain does not have an IPv6 address and getaddrinfo raises an error if I try to resolve one:

```
>>> socket.getaddrinfo('www.the-playground.dk', None, socket.AF_INET6)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.gaierror: [Errno -2] Name or service not known
```

**Comment by zachriggle** [comment]
Pretty sure this was fixed by bb5ea7677c0ba810d34148acf7c7ac927a914b0e.

This bug was opened on May 21, that fix was uploaded May 22.

**Comment by zachriggle** [comment]
Which was part of https://github.com/Gallopsled/pwntools/pull/542
